resync-from-csv.py

```diff
@@ -0,0 +1,210 @@
+            github_for_jira_auth=create_slauth(env))
+    elif env == 'staging':
+        return Environment(
+            github_for_jira_url='https://github-for-jira.us-west-1.staging.atl-paas.net',
+            github_for_jira_auth=create_slauth(env))
+    elif env == 'prod':
+        return Environment(
+            github_for_jira_url='https://github-for-jira.us-west-1.prod.atl-paas.net',
+            github_for_jira_auth=create_slauth(env))
+    else:
+        raise ValueError(f'Invalid environment {env}')
+
+@dataclass(frozen=True)
+class Installation:
+    installationId: str
+
+    @staticmethod
+    def from_dict(d):
+        return Installation(d['installation_id'])
+
+def process_installation(env: Environment, installationsString) -> bool:
+    url = '{}/api/resync'.format(env.github_for_jira_url)
+
+#   Convert the string of ids to a actual integer list
+    installations = map(lambda x: int(x), installationsString.split(','))
+
+    LOG.debug('Starting rotation of %s with url: %s', installations, url)
+    response = session.post(url,
+      auth=env.github_for_jira_auth,
+      json={
+        "installationIds": list(installations),
+        "syncType": "full",
+        "statusTypes": ["FAILED", "PENDING", "ACTIVE", "COMPLETE"],
+        "targetTasks": ["pull"]
+      },
+      headers = {"Content-Type": "application/json"})
+
+    if response.ok:
+        LOG.info('Sync of %s successfully started: (%s).',
+                 installations, response.status_code)
+        return 'success'
+    else:
+        LOG.error('Sync of %s failed (%s): %s',
+                  installations, response.status_code, response.text)
+        return 'error'
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--env', choices=('dev', 'staging', 'prod'), required=True)
+    parser.add_argument('--batchsize', default=10, type=float, help='How many installations to process each iteration')
+    parser.add_argument('--input', type=argparse.FileType('r'), required=True)
+    parser.add_argument('--output', type=argparse.FileType('a+'), required=True)
+    parser.add_argument('--sleep', type=float, help='How long to wait between requests in seconds', required=True)
+    parser.add_argument('--level', default='INFO', choices=('INFO', 'ERROR', 'WARNING', 'DEBUG', 'CRITICAL'))
+    args = parser.parse_args()
+
+    configure_logging(level=args.level)
+
+    env = create_environment(args.env)
+    input_file = args.input
+    output_file = args.output
+
+    processed = set()
+    output_file.seek(0)
+    output_reader = csv.DictReader(output_file, fieldnames=['installation_id'])
+    for row in output_reader:
+        processed.add(Installation.from_dict(row))
+    output_writer = csv.DictWriter(output_file, fieldnames=['installation_id', 'status'])
+    input_reader = csv.DictReader(input_file, fieldnames=['installation_id'])
+
+    batch = []
+    batchRow = ''
+    batchCount = 0
+
+    for row in input_reader:
```

Comment:
The batching code in this section is pretty cumbersome, could do with some proper refactoring but it works, and can improve the script over time.... was my thinking. But aware of the inefficiencies.